confgenerator/confgenerator_test.go

```diff
@@ -109,71 +107,29 @@ var (
 func TestGoldens(t *testing.T) {
 	t.Parallel()
 
-	// Iterate platforms inside test directories so the test name hierarchy matches the testdata hierarchy.
-	for _, test := range []struct {
-		dirName      string
-		errAssertion func(t *testing.T, err error, got map[string]string)
-	}{
-		{
-			validTestdataDirName,
-			func(t *testing.T, err error, got map[string]string) {
-				assert.NilError(t, err)
-				delete(got, builtinConfigFileName)
-			},
-		},
-		{
-			invalidTestdataDirName,
-			func(t *testing.T, err error, got map[string]string) {
-				assert.Assert(t, err != nil, "expected test config to be invalid, but was successful")
-				// Error is checked by runTestsInDir
-				delete(got, builtinConfigFileName)
-			},
-		},
-		{
-			builtinTestdataDirName,
-			nil,
-		},
-	} {
-		test := test
-		t.Run(test.dirName, func(t *testing.T) {
-			t.Parallel()
-			for _, pc := range testPlatforms {
-				pc := pc
-				t.Run(pc.name, func(t *testing.T) {
-					t.Parallel()
-					runTestsInDir(
-						t,
-						pc,
-						test.dirName,
-						test.errAssertion,
-					)
-				})
-			}
-		})
-	}
-}
-
-func runTestsInDir(
-	t *testing.T,
-	pc platformConfig,
-	testTypeDir string,
-	errAssertion func(*testing.T, error, map[string]string),
-) {
-	platformTestDir := filepath.Join(testTypeDir, pc.name)
-	testNames := getTestsInDir(t, platformTestDir)
+	goldensDir := "goldens"
+	testNames := getTestsInDir(t, goldensDir)
 
 	for _, testName := range testNames {
 		// https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables
 		testName := testName
 		t.Run(testName, func(t *testing.T) {
 			t.Parallel()
-			testDir := filepath.Join(platformTestDir, testName)
-			got, err := generateConfigs(pc, testDir)
-			if errAssertion != nil {
-				errAssertion(t, err, got)
-			}
-			if err := testGeneratedFiles(t, got, testDir); err != nil {
-				t.Errorf("Failed to check generated configs: %v", err)
+			for _, pc := range testPlatforms {
+				pc := pc
+				t.Run(pc.name, func(t *testing.T) {
+					testDir := filepath.Join(goldensDir, testName)
+					got, err := generateConfigs(pc, testDir)
+					if strings.HasPrefix(testName, "invalid-") {
+						assert.Assert(t, err != nil, "expected test config to be invalid, but was successful")
+					}
+					if testName != "builtin" {
+						delete(got, builtinConfigFileName)
+					}
```

Comment:
Is this still needed?

Reply:
Yes, this is needed to avoid having a duplicate golden built-in config file in every single test directory.